src/App.tsx

```diff
@@ -1,61 +1,140 @@
+import { getNewTodoID } from './services/todoService';
+
+const todosWithUser: TodoWithUser[] = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: findUserById(todo.userId),
+  };
+});
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosWithUser);
+
+  const [reloadForm, setReloadForm] = useState(0);
+
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const onSubmit = (newTodo: TodoWithUser) => {
+    return setTodos(currentTodos => [...currentTodos, newTodo]);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleOnSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title) {
+      setHasTitleError(true);
+    }
+
+    if (!userId) {
+      setHasUserIdError(true);
+    }
+
+    if (!userId || !title) {
+      return;
+    }
+
+    onSubmit({
+      id: getNewTodoID(todos),
+      title,
+      completed: false,
+      userId: userId,
+      user: findUserById(userId),
+    });
+
+    setReloadForm(reloadForm + 1);
+
+    reset();
+  };
+
+  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value.trimStart());
+    setHasTitleError(false);
+  };
+
+  const handleChooseName = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form
+        action="/api/todos"
+        method="POST"
+        key={reloadForm}
+        onSubmit={handleOnSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            {'Title: '}
+            <input
+              type="text"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={title}
+              onChange={handleChangeTitle}
+            />
+          </label>
+
+          {hasTitleError && <span className="error">Please enter a title</span>}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>
-              Choose a user
-            </option>
-          </select>
+          <label>
+            {'User '}
+            <select
+              data-cy="userSelect"
+              onChange={handleChooseName}
+              value={userId}
+            >
+              {userId ? (
+                <option value={0} disabled>
+                  Choose a user
+                </option>
+              ) : (
+                <option value={0}>Choose a user</option>
+              )}
```

Comment:
Only `disabled` attribute is changing. And actually you can use only option with disabled in any case